### PR TITLE
create a preloader

### PR DESCRIPTION
**Pull request by sweetcuptea**
Create a preloader to give a loading-like appearance every time the website is loaded.

First, I created it by making a Preloader.vue component, then I provided props that will receive numerical values from the parent element and create a div with a dynamic style width value from props that have been given

On the App.vue as the parent element I imported Preloader.vue and created some conditional rendering when the Preloader component has finished the loading animation then the main element will be rendered

The loading that I create is based on the `progress` and `preload` state that I use.
The `progress` state is used to send props to the Preloader component while the `preload` state is to determine the conditional rendering

I used both states in the `mounted()` lifecycle hook and created an interval that contains an if statement. If progress is less than 100, it will increase by 1 every 10 milliseconds. If the condition is met, it will clear the interval and set the `preload` state to false. In this way, the `main` element will be rendered

Also, I changed the barbel.ico to favicon.ico the difference between both files is that barbel.ico doesn't have a transparent background which makes the icon is ugly

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,12 +1,41 @@
...
+        clearInterval(preloadInterval);
+        this.preload = false;
+      }
+    }, 10);
+  },
+};
 </script>
 
 <style lang="scss" scoped>
+main {
```
Comment (trihargianto) [inline src/App.vue:38]: please use class. Reference: https://v2.vuejs.org/v2/style-guide/?redirect=true#Element-selectors-with-scoped-use-with-caution

File: src/App.vue [hunk 2]
```diff
@@ -1,12 +1,41 @@
...
   </div>
 </template>
 
 <script>
-export default {}
+import Preloader from "./assets/components/atoms/Preloader.vue";
+
+export default {
+  components: {
+    "tx-preloader": Preloader,
```
Comment (trihargianto) [inline src/App.vue:16]: what is `tx` ?

File: src/App.vue [hunk 3]
```diff
@@ -1,12 +1,41 @@
...
+  },
+  data() {
+    return {
+      preload: true,
```
Comment (trihargianto) [inline src/App.vue:20]: please use Boolean convention. Reference: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

File: src/App.vue [hunk 4]
```diff
@@ -1,12 +1,41 @@
...
+      progress: 0,
+    };
+  },
+  mounted() {
+    const preloadInterval = setInterval(() => {
+      if (this.progress < 100) {
+        this.progress++;
+      } else {
+        clearInterval(preloadInterval);
```
Comment (trihargianto) [inline src/App.vue:29]: is this working?

File: src/App.vue (hunk 1 above)
Comment (sweetcuptea) [reply src/App.vue:38]: thank you for the advice, soon will be fixed

File: src/App.vue (hunk 2 above)
Comment (trihargianto) [reply src/App.vue:16]: Ok then rename your Placeholder.vue component become `TxPlaceholder`

Comment (trihargianto) [reply src/App.vue:16]: don't differentiate the component naming like this

```
components: {
  "txt-preloader": Preloader
}
```

Do this instead

```
components: {
  TxPreloader
}
```

Comment (sweetcuptea) [reply src/App.vue:16]: got it ✔️

File: src/App.vue (hunk 3 above)
Comment (sweetcuptea) [reply src/App.vue:20]: am going to change it to `isLoad`, does that sound good?

Comment (trihargianto) [reply src/App.vue:20]: `isLoading` better. present tense = (is/am/are) + (verb+ing)

File: src/App.vue (hunk 4 above)
Comment (sweetcuptea) [reply src/App.vue:29]: yes, it is working fine.
does this code feel weird? or is there any other method that you know to do this correctly?

File: src/App.vue (hunk 3 above)
Comment (sweetcuptea) [reply src/App.vue:20]: okay, thank you for the explanation

File: src/App.vue (hunk 4 above)
Comment (trihargianto) [reply src/App.vue:29]: i'll back to this later

Comment (trihargianto) [reply src/App.vue:29]: this is worked well. please continue

File: src/assets/components/atoms/Preloader.vue [hunk 5]
```diff
@@ -0,0 +1,46 @@
+t
```
Comment (trihargianto) [inline src/assets/components/atoms/Preloader.vue:1]:
```suggestion

```

**Comment by trihargianto** [comment]
I think it's better to make your app structure like this

```
├── src
│   ├── assets
│   │   ├── img/
│   │   └── # other files ...
│   ├── components/
│   │   ├── atoms/
│   │   ├── molecules/
│   │   └── organisms/
│   └── ...
└── ...
```

It's common to differentiate "components" and "assets"

File: src/assets/components/atoms/Preloader.vue (hunk 5 above)
Comment (sweetcuptea) [reply src/assets/components/atoms/Preloader.vue:1]: got it ✔️

**Review by trihargianto — APPROVED** [review]
Looks Mantap from Me ✅

**Pull request merged**